models/lobby.go

```diff
@@ -185,18 +185,35 @@ func (lobby *Lobby) KickAndBanPlayer(player *Player) *helpers.TPError {
 }
 
 func (lobby *Lobby) ReadyPlayer(player *Player) *helpers.TPError {
-	// TODO implement
+	slot := &LobbySlot{}
+	err := db.DB.Where("lobby_id = ? AND player_id = ?", lobby.ID, player.ID).First(slot).Error
+	if err != nil {
+		return helpers.NewTPError("Player is not in the lobby.", 5)
+	}
+	slot.Ready = true
+	lobby.Save()
```

Comment:
you should save slot: `db.DB.Save(slot)`. You've changed the LobbySlot object that's different from the Lobby object, and since you haven't changed lobby, lobby.Save() will do nothing